modules/nodes-pods-vertical-autoscaler-tuning.adoc

```diff
@@ -20,7 +20,165 @@ You can perform the following tunings on the VPA components by editing the `Vert
+| 260Mi 
+| (0.03c)/2 + 10 ^[1]^
+| (0.1c)/2 + 50 ^[1]^
+
+s| Recommender
+| 25m
+| 100Mi
+| 50m
+| 160Mi
+| 75m
+| 275Mi
+| 120m
+| 420Mi
+| (0.05c)/2 + 50 ^[1]^
+| (0.15c)/2 + 120 ^[1]^
+
+s| Updater
+| 25m
+| 100Mi
+| 50m
+| 220Mi
+| 80m
+| 350Mi
+| 150m
+| 500Mi
+| (0.07c)/2 + 20 ^[1]^
+| (0.15c)/2 + 200 ^[1]^
+
+|===
+[.small]
+. `c` is the number of containers in the cluster.
+
+[NOTE]
+====
+It is recommended that you set the memory limit on your containers to double the recommended requests in the table. However, because CPU is a compressible resource, setting CPU limits for containers can throttle the VPA. As such, it is recommended that you do not set a CPU limit on your containers.
+====
+
+.Rate limits by VPAs in the cluster
+[cols="1,3,2,3,2,3,2,3,2"]
+|===
+| Component 2+| 1 - 150 VPAs 2+| 151 - 500 VPAs 2+| 501-2000 VPAs 2+| 2001-4000 VPAs
+
+| 
+| *QPS Limit* ^[1]^
+| *Burst* ^[2]^
+| *QPS Limit*
+| *Burst*
+| *QPS Limit*
+| *Burst*
+| *QPS Limit*
+| *Burst*
+
+s| Recommender
+| 5
+| 10
+| 30
+| 60
+| 60
+| 120
+| 120
+| 240
+
+s| Updater
+| 5
+| 10
+| 30
+| 60
+| 60
+| 120
+| 120
+| 240
+
+|===
+[.small]
+. QPS specifies the queries per second (QPS) limit when making requests to Kubernetes API server. The default is `5.0`. 
+. Burst specifies the burst limit when making requests to Kubernetes API server. The default is `10.0`.
+
+[NOTE]
+====
+* If you have more than 4000 VPAs in your cluster, note the following recommendations: 
+** It is recommended that you start performance tuning with the values in the table and slowly increase the values until you achieve the desired recommender and updater latency and performance. You should perform the performance tuning slowly because increased QPS and Burst could affect the cluster health and slow down the Kubernetes API server if too many API requests are being sent to the API server from the VPA components.
+====
+
+////
+Hiding these two NOTEs as not supported
+** It is recommended that you increase the recommender and updater interval, which is how often the VPA fetches pod metrics. However, with the longer intervals, the Operator takes longer to recommend and restart pods. The example `VerticalPodAutoscalerController` CR that follows includes the parameters to increase the intervals.
+** If you increase the `recommender-interval` value, it is recommended that you also increase the `checkpoints-timeout` value, which configures the timeout for writing VPA checkpoints after the start of the recommender interval. It is recommended that you set the timeout to the same value `recommender-interval` so that the recommender pod has time to write checkpoints before the next interval. The example `VerticalPodAutoscalerController` CR that follows includes the parameters to increase the timeout.
+////
+
+.Admission Rate limits (by custom resource pod creation surge) ^[1]^
+[options="header"]
+|===
+| Component 2+| 1-25 pod creation surge 2+| 26-50 pod creation surge 2+| 50+ pod creation surge
```

Comment:
I think we should be explicit here or in the table cells, or both that this is `custom resource pod creation surge` and not normal pod surge. So it would just be changing `pod creation surge` -> `custom resource pod creation surge` or `CR pod creation surge` if that makes sense,